Add docs for Workspaces Settings > Policies

tstormk opened the pull request:
### What
For the GA release of the new Sentinel policy evaluation, add a documentation on how users can toggle between the new and old policy evaluation experiences on a per-workspace basis.

### Screenshots
<img width="1072" alt="Screenshot 2023-05-26 at 13 16 08" src="https://github.com/hashicorp/terraform-docs-common/assets/260435/f361016c-1540-4e6d-a5b0-800a22c99056">
<img width="995" alt="Screenshot 2023-05-26 at 13 16 14" src="https://github.com/hashicorp/terraform-docs-common/assets/260435/bdd53c44-e2a8-457d-9347-972fa49bf127">


----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] N/A One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [x] N/A Description links to related pull requests or issues, if any.

#### Content
- [x] N/A Redirects have been added for moved, renamed, or deleted pages. This requires a separate PR in the [`terraform-website` repository](https://github.com/hashicorp/terraform-website) `redirects.next.js` file.
- [x] N/A API documentation and the API Changelog have been updated. 
- [x] N/A Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] N/A Pages with related content are updated and link to this content when appropriate.
- [x] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [x] New pages have metadata (page name and description) at the top.
- [x] N/A New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [x] N/A New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] N/A UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [x] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

rkoron007 commented on website/docs/cloud-docs/workspaces/settings/policies.mdx:
```diff
@@ -0,0 +1,15 @@
+---
+page_title: Policies - Workspaces - Terraform Cloud
```
@tstormk Does this need to be it's own page? Will we be adding more content later? 

If not, a whole page feels unnecessary, and we could instead add a new section in the main workspace settings page (https://github.com/hashicorp/terraform-docs-common/blob/main/website/docs/cloud-docs/workspaces/settings/index.mdx) to cover this and be good to go as @hcrhall suggested. What do you think?

rkoron007 commented on website/docs/cloud-docs/workspaces/settings/policies.mdx:
```diff
@@ -0,0 +1,15 @@
+
+# Policies
+
+Terraform Cloud provides two different experiences for Sentinel policy evaluations. You can adjust your settings to use the experience you prefer. By default, the newest policy evaluation experience is enabled.
```
Do we have a link to the new policy evaluation experience? As in, **how** does this effect a user's sentinel policy experience? Can we link to the Sentinel docs somewhere or provide an example?

Suggestions for style:
```suggestion
Terraform Cloud offers two experiences for Sentinel policy evaluations. You can adjust your **Sentinel Experience** settings to your preferred experience. By default, Terraform Cloud enables the newest policy evaluation experience.
```

rkoron007 commented on website/docs/cloud-docs/workspaces/settings/policies.mdx:
```diff
@@ -0,0 +1,15 @@
+
+# Toggling Between Sentinel Policy Experiences
+
+To toggle between the two Sentinel policy evaluation experiences, click the **Enable the new Sentinel policy experience** toggle under the **Sentinel Experience** heading. Your changes are persisted automatically.
```
Suggestion for active voice:
```suggestion
To toggle between the two Sentinel policy evaluation experiences, click the **Enable the new Sentinel policy experience** toggle under the **Sentinel Experience** heading. Terraform Cloud persists your changes automatically. If Terraform Cloud is performing a run on a different page, you must refresh that page to see changes to your policy evaluation experience.
```

rkoron007 commented on website/docs/cloud-docs/workspaces/settings/policies.mdx:
```diff
@@ -0,0 +1,15 @@
+
+To toggle between the two Sentinel policy evaluation experiences, click the **Enable the new Sentinel policy experience** toggle under the **Sentinel Experience** heading. Your changes are persisted automatically.
+
+-> **Note:** If you have a run with a Sentinel policy evaluation open in a different tab, you will need to refresh that page in under to see the changes to the policy evaluation experience.
```
We try to avoid call outs unless absolutely necessary, so suggesting we add this sentence to the above paragraph:
```suggestion
```

rkoron007 reviewed (commented):
Left some comments; please feel free to push back on anything I suggested!

tstormk commented:
I addressed your comments @rkoron007 - thanks for the feedback! I've also updated the screenshots in the PR body.

I don't know if the new `Policies` section is too long compared to the other sections on the page. I figured I would add all the text from the "old" page, because there is no longer form documentation we can link to like you asked. I don't think there was such a piece of documentation on the old policy evaluation experience either, because the only other changes made to the policy evaluation experience documentation have been [changing some screenshots to show the new experience instead of the old](https://github.com/hashicorp/tutorials/pull/1264).

Should we add a page for documenting the policy evaluation experiences or should we keep it as is? Perhaps add some screenshots to the new `Policies` section?

rkoron007 reviewed (approved):
This looks great to me!
